tests/pipeline/test_pipeline.py

```diff
@@ -323,6 +323,27 @@ def test_combine(self):
         assert new_pipeline.outputs() == {"output1", "output2"}
         assert {n.name for n in new_pipeline.nodes} == {"a", "b"}
 
+    def test_remove(self):
```

Comment:
Can you add a couple of test cases where you remove a pipeline that has no nodes in common, or respectively only *some* nodes in common? I'm thinking nothing should happen in the first case (p1 - p2 = p1), and some nodes should go away in the second case, like in set algebra.

Reply:
I have added some tests for empty pipelines, and pipelines with partial intersections.  Thanks for the suggestion this will cover more edge cases.